### ISSUE TITLE
Deal with NaN loss issue

### ISSUE THREAD

**Issue by vollOlga** [GeroVanMi/algorithmic-quartet-mlops#28]

**Comment by GeroVanMi** [comment]
Note: The current training code causes the model to quickly produce `NaN` loss. I assume that this could be due to exploding gradients (lack of BatchNormalization?)

**Comment by GeroVanMi** [comment]
Nevermind, this is due to an issue with the GTX 1660 Graphics Card series not being able to handle 16 bit floating point precision: https://github.com/pytorch/pytorch/issues/58123

Loss calculation works as intended now when using 32 bit fp.
